server: TestStatusEngineStatsJson failed

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8366707?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8366707?buildTab=artifacts#/) on master @ [942b55ef3f329d2e5e8142c8fc5282ed56173ea7](https://github.com/cockroachdb/cockroach/commits/942b55ef3f329d2e5e8142c8fc5282ed56173ea7):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/04ff595e8b8a4cdf61532aa9937d7edf/logTestStatusEngineStatsJson2437800523
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:278: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/04ff595e8b8a4cdf61532aa9937d7edf/logTestStatusEngineStatsJson2437800523
--- FAIL: TestStatusEngineStatsJson (8.45s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23546

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8577071?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8577071?buildTab=artifacts#/) on master @ [1708ea4b2a2b36bb88c028b6db65b023b28a73cf](https://github.com/cockroachdb/cockroach/commits/1708ea4b2a2b36bb88c028b6db65b023b28a73cf):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5dd3194da4351b03928fb98be247cd8e/logTestStatusEngineStatsJson1936360921
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:281: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5dd3194da4351b03928fb98be247cd8e/logTestStatusEngineStatsJson1936360921
--- FAIL: TestStatusEngineStatsJson (7.24s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8634439?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8634439?buildTab=artifacts#/) on master @ [284d9de2d12dc8b85c463b5b7654e0cb1f7c7dc7](https://github.com/cockroachdb/cockroach/commits/284d9de2d12dc8b85c463b5b7654e0cb1f7c7dc7):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson1514256422
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:281: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson1514256422
--- FAIL: TestStatusEngineStatsJson (42.76s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8634439?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8634439?buildTab=artifacts#/) on master @ [284d9de2d12dc8b85c463b5b7654e0cb1f7c7dc7](https://github.com/cockroachdb/cockroach/commits/284d9de2d12dc8b85c463b5b7654e0cb1f7c7dc7):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson3122952835
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:281: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson3122952835
--- FAIL: TestStatusEngineStatsJson (28.70s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8706196?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8706196?buildTab=artifacts#/) on master @ [494b909193267b9810f181187a9e57207d81f930](https://github.com/cockroachdb/cockroach/commits/494b909193267b9810f181187a9e57207d81f930):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson1597719429
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:281: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: Error while dialing connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/290373a27074403e036851126b3225f9/logTestStatusEngineStatsJson1597719429
--- FAIL: TestStatusEngineStatsJson (17.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9138997?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9138997?buildTab=artifacts#/) on master @ [9aa4f3485c75397e9ce0edabbf22c6343e1a901a](https://github.com/cockroachdb/cockroach/commits/9aa4f3485c75397e9ce0edabbf22c6343e1a901a):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8eac222af22ad63c16ed43335cd017c2/logTestStatusEngineStatsJson2471064280
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:281: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8eac222af22ad63c16ed43335cd017c2/logTestStatusEngineStatsJson2471064280
--- FAIL: TestStatusEngineStatsJson (16.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStatusEngineStatsJson [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9348592?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9348592?buildTab=artifacts#/) on master @ [7074da619b850a279a70d6f43ffd1aac4cc5fe67](https://github.com/cockroachdb/cockroach/commits/7074da619b850a279a70d6f43ffd1aac4cc5fe67):


```
=== RUN   TestStatusEngineStatsJson
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/74885ac1e7f972ae9e97608ed3fdb458/logTestStatusEngineStatsJson733538378
    test_log_scope.go:79: use -show-logs to present logs inline
    status_test.go:282: status: 503 Service Unavailable, content-type: application/json, body: {
          "error": "connection error: desc = \"transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "code": 14,
          "message": "connection error: desc = \"transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)\"",
          "details": [
          ]
        }, error: <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/74885ac1e7f972ae9e97608ed3fdb458/logTestStatusEngineStatsJson733538378
--- FAIL: TestStatusEngineStatsJson (169.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99261 server: TestStatusEngineStatsJson failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusEngineStatsJson.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/status_test.go
+++ b/pkg/server/status_test.go
@@ -278,9 +278,14 @@ func TestStatusEngineStatsJson(t *testing.T) {
 	defer s.Stopper().Stop(context.Background())
 
 	var engineStats serverpb.EngineStatsResponse
-	if err := getStatusJSONProto(s, "enginestats/local", &engineStats); err != nil {
-		t.Fatal(err)
-	}
+	// Using SucceedsSoon because we have seen in the wild that
+	// occasionally requests don't go through with error "transport:
+	// error while dialing: connection interrupted (did the remote node
+	// shut down or are there networking issues?)"
+	testutils.SucceedsSoon(t, func() error {
+		return getStatusJSONProto(s, "enginestats/local", &engineStats)
+	})
+
 	if len(engineStats.Stats) != 1 {
 		t.Fatal(errors.Errorf("expected one engine stats, got: %v", engineStats))
 	}

